### DISCUSSION TITLE
Do I have to provide com.squareup.okio dependency in pom.xml?

### DISCUSSION THREAD

**Discussion by domfe** [fabric8io/kubernetes-client#6440] [Q&A] [answered]
Hello.
I have an error when using openshift-client
 "OkHttp Dispatcher" java.lang.NoSuchMethodError: 'boolean okio.ByteString.startsWith(okio.ByteString)'

I was able to resolve the issue by explicitly inserting the dependency of okio library in my pom.xml.
Is it the correct approach or it should be resolved automatically by fabric8?

<dependency>
	<groupId>io.fabric8</groupId>
	<artifactId>openshift-client</artifactId>
</dependency>

Thank you.

**Comment by manusa** [comment]
It should be automatically resolved.
Are you using a different version of OkHttp or do you have an explicit dependency to OkHttp?
Could you please provide a trimmed version of your pom.xml so we can reproduce?

**Reply by domfe** [reply]
No, I don't have direct reference to OkHttp. My project is a quarkus project. Should I switch to [kubernetes-httpclient-vertx](https://mvnrepository.com/artifact/io.fabric8/kubernetes-httpclient-vertx)

[pom-extract.txt](https://github.com/user-attachments/files/17343117/pom-extract.txt)

**Reply by manusa** [reply, marked as answer]
You should be using the openshift-client-extension instead of referencing the fabric8 library directly:
https://quarkus.io/extensions/io.quarkus/quarkus-openshift-client/

**Reply by domfe** [reply]
No dependency problem with openshift-client-extension.

Thank you.
